Fabric.Authorization.API/scripts/AzureAD-Migration.ps1

```
@@ -0,0 +1,252 @@
+                }
+
+                Write-DosMessage -Level "Information" -Message "Retrieving group $($groupSID) from Azure AD Tenant $($tenantId)..."
+                $azureADGroups = Get-AzureADGroup -Filter "onPremisesSecurityIdentifier eq '$($groupSID)'"
+                $authenticationFailed = $false
+            }
+            catch [Microsoft.Open.Azure.AD.CommonLibrary.AadAuthenticationFailedException], [Microsoft.Open.Azure.AD.CommonLibrary.AadNeedAuthenticationException] {
+                Write-DosMessage -Level Error -Message "Error authenticating user with Azure AD tenant $($tenantId). Please try again."
+                $authenticationFailed = $true
+            }
+            # Connect-AzureADTenant will not throw an exception if a valid username (i.e., user@domain, even if user is not in tenant
+            # and/or invalid password is entered). This catch block handles the 403 that is returned when Get-AzureADGroup fails
+            # for this scenario
+            catch [Microsoft.Open.AzureAD16.Client.ApiException] {
+                if ($_.Exception.ErrorCode -eq 403) {
+                    Write-DosMessage -Level Information -Message "403 error when retrieving Azure AD group."
+                    # reset cached credentials for this tenantId
+                    $credentials.Remove($tenantId)
+                    Write-DosMessage -Level "Information" -Message "Removed cached credentials for Azure AD Tenant $($tenantId)..."
+                    $authenticationFailed = $true
+                }
+                else {
+                    Write-DosMessage -Level Information -Message "Unexpected error when retrieving Azure AD group. $($_.Exception)"
+                    throw
+                }
+            }
+            catch {
+                $errorMsg = "Unexpected error while connecting to Azure AD: $($_.Exception)"
+                Write-DosMessage -Level Error -Message $errorMsg
```

If we want to throw the error, we can simply use Write-DosMessage -Level Fatal -Message $errorMsq